Makefile

```diff
@@ -210,20 +214,23 @@ endif
 $(HOME)/go/bin/mockery:
 	go get github.com/vektra/mockery/.../
 
+dist/update-mocks: $(HOME)/go/bin/mockery $(MOCK_FILES)
+	./hack/update-mocks.sh $(MOCK_FILES)
+	touch dist/update-mocks
+
 .PHONY: codegen
-codegen: $(HOME)/go/bin/mockery
-	# Generate code
+codegen: dist/update-mocks
+	# Back-up go.*, but only if we have not already done this (because that would suggest we failed mid-codegen and the currenty go.* files are borked).
+	[[ -e dist/go.mod ]] || cp go.mod go.sum dist/
```

Comment:
this change is intended to stop the codegen job being red due to golang/tools changes